tests/lib/rules/no-param-reassign.js

```diff
@@ -30,7 +30,8 @@ ruleTester.run("no-param-reassign", rule, {
         { code: "function foo(a) { [a.b] = []; }", parserOptions: { ecmaVersion: 6 } },
         { code: "function foo(a) { bar(a.b).c = 0; }", options: [{ props: true }] },
         { code: "function foo(a) { data[a.b] = 0; }", options: [{ props: true }] },
-        { code: "function foo(a) { +a.b; }", options: [{ props: true }] }
+        { code: "function foo(a) { +a.b; }", options: [{ props: true }] },
+        { code: "function foo(a) { a.b = 0; }", options: [{ props: true, ignorePropertyAssignmentsFor: ["a"] }] }
```

Comment:
Could you add a couple more tests as a sanity check?

* A property reassignment for a parameter that does *not* appear in `ignorePropertyAssignmentsFor` should cause an error to be reported.
* If `ignorePropertyAssignmentsFor` contains multiple elements, property assignment should also be ignored for elements other than the first element in the list.

I think these tests should already pass with the current implementation, but they might be useful to have if we end up changing the implementation of the rule in the future.